File: MyClinicTask/Domain/Entities/Employee.cs
using System;

namespace MyClinicTask.Domain.Entities
{
    public class Employee
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public int NationalityId { get; set; }
        public virtual Nationality Nationality { get; set; }
        public int PositionId { get; set; }
        public virtual Position Position { get; set; }
        public double Salary { get; set; }
    }
}

File: MyClinicTask/ClientApp/src/app/features/employees/models/employee.ts
import { NationalityDto, PositionDto } from "./lookup";

export interface EmployeesVm {
    list: EmployeeDto[],
    recordsCount: number
}

export interface EmployeeVm {
    item: EmployeeDto
}

export interface EmployeeDto {
    id: number,
    firstName: string,
    middleName: string,
    lastName: string,
    dateOfBirth: Date,
    nationality: NationalityDto,
    position: PositionDto,
    salary: number
}

export interface PostEmployeeCmd {
    firstName: string,
    middleName: string,
    lastName: string,
    dateOfBirth: string,
    nationalityId: number,
    positionId: number,
    salary: number
}